archive/progress_http/src/test/scala/uk/ac/wellcome/platform/archive/progress_http/ProgressHttpFeatureTest.scala

```diff
@@ -216,5 +285,353 @@ class ProgressHttpFeatureTest
+                 |    "id": "create",
+                 |    "type": "IngestType"
+                 |  },
+                 |  "sourceLocation":{
+                 |    "type": "Location",
+                 |    "provider": {
+                 |      "type": "Provider",
+                 |      "id": "${StandardDisplayProvider.id}"
+                 |    },
+                 |    "path": "b22454408.zip"
+                 |  },
+                 |  "space": {
+                 |    "id": "bcnfgh",
+                 |    "type": "Space"
+                 |  }
+                 |}""".stripMargin
+            )
+
+            val request = HttpRequest(
+              method = POST,
+              uri = url,
+              headers = Nil,
+              entity = entity
+            )
+
+            whenRequestReady(request) { response: HttpResponse =>
+              response.status shouldBe StatusCodes.BadRequest
+              response.entity.contentType shouldBe ContentTypes.`application/json`
+
+              val progressFuture =
+                Unmarshal(response.entity).to[ErrorResponse]
+
+              whenReady(progressFuture) { actualError =>
+                actualError shouldBe ErrorResponse(
+                  400,
+                  "Invalid value at .sourceLocation.bucket: required property not supplied.",
+                  "Bad Request",
+                  "Error")
+                val requests =
+                  listMessagesReceivedFromSNS(topic).map(messageInfo =>
+                    fromJson[IngestBagRequest](messageInfo.message).get)
+
+                requests shouldBe empty
+              }
+            }
+          }
+      }
+    }
+
+    it("returns a json error if the sourceLocation has an invalid bucket field") {
+      withConfiguredApp {
+        case (_, topic, baseUrl) =>
+          withMaterializer { implicit materialiser =>
+            val url = s"$baseUrl/progress"
+
+            val entity = HttpEntity(
+              ContentTypes.`application/json`,
+              s"""|{
+                  |  "type": "Ingest",
+                  |  "ingestType": {
+                  |    "id": "create",
+                  |    "type": "IngestType"
+                  |  },
+                  |  "sourceLocation":{
+                  |    "type": "Location",
+                  |    "provider": {
+                  |      "type": "Provider",
+                  |      "id": "${StandardDisplayProvider.id}"
+                  |    },
+                  |    "bucket": {"name": "bucket"},
+                  |    "path": "b22454408.zip"
+                  |  },
+                  |  "space": {
+                  |    "id": "bcnfgh",
+                  |    "type": "Space"
+                  |  }
+                  |}""".stripMargin
+            )
+
+            val request = HttpRequest(
+              method = POST,
+              uri = url,
+              headers = Nil,
+              entity = entity
+            )
```

Comment:
Not for this PR, but I wonder if we should push this repeated POST request code into a fixture?